src/front/zsharpcurly/mod.rs

```diff
@@ -957,37 +1035,21 @@ impl<'ast> ZGen<'ast> {
             .unwrap_or_else(|| panic!("No function '{}'", &f_name))
             .clone();
         // XXX(unimpl) tuple returns not supported
-        assert!(f.returns.len() <= 1);
         if !f.generics.is_empty() {
             self.err("Entry function cannot be generic. Try adding a wrapper function that supplies an explicit generic argument.", &f.span);
         }
         // get return type
-        let ret_ty = f.returns.first().map(|r| self.type_(r));
+        let ret_ty = f.return_type.map(|r| self.type_(&r));
         // set up stack frame for entry function
         self.circ_enter_fn(n.to_owned(), ret_ty.clone());
-        let mut persistent_arrays: Vec<String> = Vec::new();
+        let persistent_arrays: Vec<String> = Vec::new();
         for p in f.parameters.iter() {
             let ty = self.type_(&p.ty);
             debug!("Entry param: {}: {}", p.id.value, ty);
-            let md = self.interpret_array_md(&p.array_metadata);
+            // XXX(unimpl) array metadata
```

Comment:
Would it be better to interpret the array metadata and panic if it is non-empty? Silently ignoring it seems like a bad idea.

Reply:
In the [new grammar](https://github.com/StefanosChaliasos/circ/blob/zsharpcurly/third_party/ZoKratesCurly/zokrates_parser/src/zokrates.pest), metadata is not supported, so I guess we should skip it. Should I remove this comment from here?

Reply:
Ah! I understand now. Yes, having metadata give a parsing error is fine. You can remove the comment or not, whichever you prefer.